src/leo_redundant_manager_api.erl

```diff
@@ -597,6 +610,34 @@ get_redundancies_by_addr_id(Method, AddrId) ->
     {ok, Options} = get_options(),
     get_redundancies_by_addr_id_1(ring_table(Method), AddrId, Options).
 
+-spec(get_redundancies_by_addr_id(Method, AddrId, ConsistencyLevel) ->
+             {ok, #redundancies{}} |
+             {error, any()} when Method::method(),
+                                 AddrId::integer(),
+                                 ConsistencyLevel::[{ConsistencyItem, ConsistencyValue}],
+                                 ConsistencyItem::consistency_item(),
+                                 ConsistencyValue::non_neg_integer()).
+get_redundancies_by_addr_id(Method, AddrId, ConsistencyLevel) ->
+    {ok, Options} = get_options(),
+    N_Value = leo_misc:get_value(?PROP_N, Options),
+    BitOfRing = leo_misc:get_value(?PROP_RING_BIT, Options),
+    MDCR_N_Value = leo_misc:get_value(?PROP_N, ConsistencyLevel),
+
+    case (N_Value >= MDCR_N_Value) of
+        true ->
+            MDCR_R_Value = leo_misc:get_value(?PROP_R, ConsistencyLevel),
+            MDCR_W_Value = leo_misc:get_value(?PROP_W, ConsistencyLevel),
+            MDCR_D_Value = leo_misc:get_value(?PROP_D, ConsistencyLevel),
+            Options_1 = [{?PROP_N, MDCR_N_Value},
+                         {?PROP_R, MDCR_R_Value},
+                         {?PROP_W, MDCR_W_Value},
+                         {?PROP_D, MDCR_D_Value},
+                         {?PROP_RING_BIT, BitOfRing}],
+            get_redundancies_by_addr_id_1(ring_table(Method), AddrId, Options_1);
+        false ->
+            {error, ?ERROR_INVALID_MDCR_CONFIG}
+    end.
+
```

Comment:
Out of curiosity,
Is there any reason that N_Value must be greater/equal MDCR_N_Value?
I think there might be the case where people want the number of replicas for MDC to be greater than local replicas for some reason (like the failure rate of HD on the remote cluster is high and so on) so that allowing them to set such number would cover more use cases if there is no technical restriction.